src/gctools/interrupt.cc

```diff
@@ -296,85 +292,42 @@ int global_signalTrap = 0;
-  // See https://doc.rust-lang.org/stable/core/arch/x86_64/fn._mm_setcsr.html
-  // mask all -> no fpe-exceptions
-#if defined(__i386__) || defined(__x86_64__)
-  _MM_SET_EXCEPTION_MASK(_MM_MASK_MASK);
-  if (underflow.notnilp())
-    _mm_setcsr(_mm_getcsr() & (~_MM_MASK_UNDERFLOW));
-  if (overflow.notnilp())
-    _mm_setcsr(_mm_getcsr() & (~_MM_MASK_OVERFLOW));
-  if (inexact.notnilp())
-    _mm_setcsr(_mm_getcsr() & (~_MM_MASK_INEXACT));
-  if (invalid.notnilp())
-    _mm_setcsr(_mm_getcsr() & (~_MM_MASK_INVALID));
-  if (divide_by_zero.notnilp())
-    _mm_setcsr(_mm_getcsr() & (~_MM_MASK_DIV_ZERO));
-  if (denormalized_operand.notnilp())
-    _mm_setcsr(_mm_getcsr() & (~_MM_MASK_DENORM));
-#elif defined(CLASP_APPLE_SILICON)
-  std::fenv_t env;
-  std::fegetenv(&env);
-  env.__fpcr = (underflow.notnilp() ? __fpcr_trap_underflow : 0) | (overflow.notnilp() ? __fpcr_trap_overflow : 0) |
-               (inexact.notnilp() ? __fpcr_trap_inexact : 0) | (invalid.notnilp() ? __fpcr_trap_invalid : 0) |
-               (divide_by_zero.notnilp() ? __fpcr_trap_divbyzero : 0) | (denormalized_operand.notnilp() ? __fpcr_trap_denormal : 0);
-  std::fesetenv(&env);
-#else
-  printf("%s:%d:%s Add support for FPE masks for this architecture\n", __FILE__, __LINE__, __FUNCTION__);
-#endif
+CL_DEFUN int core__fe_disable_except(int ex) {
+  feclearexcept(FE_ALL_EXCEPT);
+  int prev = fedisableexcept(ex);
+  _lisp->setTrapFpeBits(fegetexcept());
+  return prev;
 }
 
 DOCGROUP(clasp);
-CL_DEFUN core::Fixnum_sp core__get_current_fpe_mask() {
-#if defined(__i386__) || defined(__x86_64__)
-  unsigned int before = _MM_GET_EXCEPTION_MASK();
-  return core::clasp_make_fixnum(before);
-#elif defined(__aarch64__)
-  std::fenv_t env;
-  std::fegetenv(&env);
-  return core::clasp_make_fixnum(env.__fpcr);
-#else
-  printf("%s:%d:%s Add support for FPE masks for this architecture\n", __FILE__, __LINE__, __FUNCTION__);
-  abort();
-#endif
+CL_DEFUN int core__fe_get_except() {
+  return fegetexcept();
 }
 
 DOCGROUP(clasp);
-CL_DEFUN void core__set_current_fpe_mask(core::Fixnum_sp mask) {
-  Fixnum value = core::unbox_fixnum(mask);
-#if defined(__i386__) || defined(__x86_64__)
-  _MM_SET_EXCEPTION_MASK(value);
-#elif defined(__aarch64__)
-  std::fenv_t env;
-  std::fegetenv(&env);
-  env.__fpcr = value;
-  std::fesetenv(&env);
-#else
-  printf("%s:%d:%s Add support for FPE masks for this architecture\n", __FILE__, __LINE__, __FUNCTION__);
-#endif
+CL_DEFUN int core__fe_restore_except(int ex) {
+  feclearexcept(FE_ALL_EXCEPT);
+  int prev = feenableexcept(ex);
+  fedisableexcept(~ex);
+  _lisp->setTrapFpeBits(fegetexcept());
+  return prev;
 }
 
 void handle_fpe(int signo, siginfo_t* info, void* context) {
   (void)context; // unused
-  // printf("Enter handle_fpe Signo: %d Errno:%d Code:%d\n", (info->si_signo), (info->si_errno), (info->si_code));
-  // init_float_traps(); // WHY
+  if (_lisp) {
+    feenableexcept(_lisp->getTrapFpeBits());
```

Comment:
The the interrupts should get queued. On amd64 the traps are all reset on SIGFPE so you have to restore them. This is not true on Aarch64.

Reply:
The `if (_lisp)` check is just there because I am paranoid.

Reply:
Ok, could you mention the reset in a comment? Also, do you have a doc reference for this or is it just what you've observed.

Reply:
Just what I have observed, but I'll look for a reference. I think it might originate with how FPE is enabled/disabled. On arm64 FPE is always on until you mask it to off and then the mask is reset. Whereas on aarch64 the FPE is off until you explicitly enable it. Aarch64 doesn't have the concept of "masking."